### PR TITLE
Suggest adding a missing semicolon before an item

### LINKED ISSUE #100533: Add missing semicolon suggestion for global variable declarations
Given the following code: [link](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=6d2d52a33240be244ddd825a53b75200)

```rust
const x: i32 = 123

fn main() {

}
```

The current output is:

```
   Compiling playground v0.0.1 (/playground)
error: expected one of `.`, `;`, `?`, or an operator, found keyword `fn`
 --> src/main.rs:3:1
  |
1 | const x: i32 = 123
  |                   - expected one of `.`, `;`, `?`, or an operator
2 |
3 | fn main() {
  | ^^ unexpected token

error: could not compile `playground` due to previous error
```

Ideally the output should suggest putting a semicolon at the end of the declaration, since that suggestion is present if that declaration is placed in the main function.

### PR DESCRIPTION
**Pull request by TaKO8Ki**
fixes #100533

### CODE REVIEW

File: compiler/rustc_ast/src/token.rs [hunk 1]
```diff
@@ -436,6 +436,30 @@ impl Token {
             || self == &OpenDelim(Delimiter::Parenthesis)
     }
 
+    /// Returns `true` if the token can appear at the start of an item.
+    pub fn can_begin_item(&self) -> bool {
+        match self.kind {
+            Ident(name, _) => [
+                kw::Fn,
```
Comment (compiler-errors) [inline compiler/rustc_ast/src/token.rs:443]: This list doesn't exist somewhere else in the parser already, right?

**Comment by compiler-errors** [comment]
r? @compiler-errors

r=me with question answered

File: compiler/rustc_ast/src/token.rs (hunk 1 above)
Comment (TaKO8Ki) [reply compiler/rustc_ast/src/token.rs:443]: No, it doesn't 👍

**Pull request merged**